crates/builder/benches/dotgraph.rs

```diff
@@ -0,0 +1,59 @@
+use criterion::{black_box, criterion_group, criterion_main, Criterion, SamplingMode};
+use graph_builder::prelude::*;
+
+mod common;
+
+use common::*;
+use graph_builder::input::dotgraph::{LabelStats, NodeLabelIndex};
+use rand::Rng;
+
+fn label_stats(c: &mut Criterion) {
+    let mut group = c.benchmark_group("label_stats");
+
+    group.sampling_mode(SamplingMode::Flat);
+
+    group.bench_function(SMALL.name, |b| bench_label_stats(b, SMALL));
+    group.bench_function(MEDIUM.name, |b| bench_label_stats(b, MEDIUM));
+    group.bench_function(LARGE.name, |b| bench_label_stats(b, LARGE));
+
+    group.finish();
+}
+
+fn bench_label_stats(b: &mut criterion::Bencher, Input { node_count, .. }: Input) {
+    let labels = common::gen::node_values(node_count, |_node, rng| rng.gen_range(0..42));
+    let graph: UndirectedCsrGraph<usize, usize> = GraphBuilder::new()
+        .edges([(0, node_count - 1)])
+        .node_values(labels.clone())
+        .build();
+
+    b.iter(|| black_box(LabelStats::from_graph(&graph)))
```

Comment:
I was _just_ listening to https://rustacean-station.org/episode/rust-1.65-1.66-1.67/ and at the 40:05 mark, black box is discussed. Jon says that in order to correctly apply this to function calls, the return value must be black boxed (as it is here), but also the function arguments. That means

```suggestion
    b.iter(|| black_box(LabelStats::from_graph(black_box(&graph))))
```

Reply:
Interesting; I'll try to adapt this to the other benchmarks as well.